src/core/core.scale.js

```diff
@@ -162,13 +182,36 @@ module.exports = function(Chart) {
 
 			// Ticks
 			me.beforeBuildTicks();
-			me.buildTicks();
+			ticks = me.buildTicks() || [];
 			me.afterBuildTicks();
 
 			me.beforeTickToLabelConversion();
-			me.convertTicksToLabels();
+			labels = me.convertTicksToLabels(ticks) || me.ticks;
```

Comment:
Yes, it's me `me.ticks` in case of a sub class that `convertTicksToLabels` in place and so modify `me.ticks` directly without returning it. I know it's pretty confusing, I will add a comment.